src/cdk/v2/destinations/bloomreach/procWorkflow.yaml

```diff
@@ -0,0 +1,119 @@
+      $.assert(.message.timestamp ?? .message.originalTimestamp, "Timestamp is not present. Aborting");
+      const userId = .message.().(
+          {{{{$.getGenericPaths("userIdOnly")}}}};
+      );
+      $.assert(userId ?? .message.anonymousId, "Either one of userId or anonymousId is required. Aborting");
+
+  - name: prepareIdentifyPayload
+    condition: $.context.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      const customerIDs = $.prepareCustomerIDs(.message, .destination);
+      const customerProperties = $.constructPayload(.message, $.CUSTOMER_PROPERTIES_CONFIG);
+      const extraCustomerProperties = $.extractCustomFields(.message, {}, ['traits', 'context.traits'], $.EXCLUSION_FIELDS);
+      const properties = {
+        ...customerProperties,
+        ...extraCustomerProperties
+      }
+      const data = .message.().({
+        "customer_ids": customerIDs,
+        "update_timestamp": $.toUnixTimestamp({{{{$.getGenericPaths("timestamp")}}}}),
+        properties
+      });
+
+      $.context.payload = $.removeUndefinedAndNullValues({name: $.CUSTOMER_COMMAND, data})
+
+  - name: prepareEventName
+    steps:
+      - name: pageEventName
+        condition: $.context.messageType === {{$.EventType.PAGE}}
+        template: |
+          const category = .message.category ?? .message.properties.category;
+          const name = .message.name || .message.properties.name;
+          const eventNameArray = ["Viewed"];
+          category ? eventNameArray.push(category);
+          name ? eventNameArray.push(name);
+          eventNameArray.push("Page");
+          $.context.event = eventNameArray.join(" ");
+      - name: screenEventName
+        condition: $.context.messageType === {{$.EventType.SCREEN}}
+        template: |
+          const category = .message.category ?? .message.properties.category;
+          const name = .message.name || .message.properties.name;
+          const eventNameArray = ["Viewed"];
+          category ? eventNameArray.push(category);
+          name ? eventNameArray.push(name);
+          eventNameArray.push("Screen");
+          $.context.event = eventNameArray.join(" ");
+      - name: trackEventName
+        condition: $.context.messageType === {{$.EventType.TRACK}}
+        template: |
+          $.assert(.message.event, "Event name is required. Aborting");
+          $.context.event = .message.event
+
+  - name: prepareTrackPageScreenPayload
+    condition: $.context.messageType !== {{$.EventType.IDENTIFY}}
+    template: |
+      const customerIDs = $.prepareCustomerIDs(.message, .destination);
+      const data = .message.().({
+        "customer_ids": customerIDs,
+        "timestamp": $.toUnixTimestamp({{{{$.getGenericPaths("timestamp")}}}}),
+        "properties": .properties,
+        "event_type": $.context.event,
+      });
+
+      $.context.payload = $.removeUndefinedAndNullValues({name: $.CUSTOMER_EVENT_COMMAND, data})
+
+  - name: buildResponse
```

Comment:
Please add reference to destination docs wherever relevent in description fields

Reply:
added doc ref in config.ts